tests/itests-cxfrs/src/test/java/org/apache/camel/kafkaconnector/cxfrs/source/CamelSourceCXFRSITCase.java

```diff
@@ -0,0 +1,331 @@
+            fail(e.getMessage());
+        }
+    }
+    
+    @Test
+    @Timeout(2000)
+    public void testBasicSendReceiveWithoutProcessor() {
+        try {
+            ConnectorPropertyFactory connectorPropertyFactory = CamelSourceCXFRSPropertyFactory
+                    .basic()
+                    .withKafkaTopic(TestUtils.getDefaultTestTopic(this.getClass()))
+                    .withAddress(CXF_RS_ENDPOINT_ADDRESS)
+                    .withResourceClass("org.apache.camel.kafkaconnector.cxfrs.CustomerServiceResource");
+                    
+                                        
+
+            runBasicStringTest(connectorPropertyFactory, false);
+        } catch (Exception e) {
+            LOG.error("CXF test failed: {}", e.getMessage(), e);
+            fail(e.getMessage());
+        }
+    }
+    
+    @Test
+    @Timeout(20)
+    public void testBasicSendReceiveUsingUrl() {
+        try {
+            ConnectorPropertyFactory connectorPropertyFactory = CamelSourceCXFRSPropertyFactory
+                    .basic()
+                    .withKafkaTopic(TestUtils.getDefaultTestTopic(this.getClass()))
+                    .withUrl(CXF_RS_ENDPOINT_URI).buildUrl();
+                    
+                    
+
+            runBasicStringTest(connectorPropertyFactory, false);
+        } catch (Exception e) {
+            LOG.error("CXF test failed: {}", e.getMessage(), e);
+            fail(e.getMessage());
+        }
+    }
+
+    
+    
+    private void invokeGetCustomer(String uri, String expect, boolean withProcessor) throws Exception {
+        HttpGet get = new HttpGet(uri);
+        get.addHeader("Accept", "application/json");
+        CloseableHttpClient httpclient = HttpClientBuilder.create().build();
+
+        try {
+            HttpResponse response = httpclient.execute(get);
+            if (withProcessor) {
+                assertEquals(200, response.getStatusLine().getStatusCode());
+                assertEquals(expect,
+                         EntityUtils.toString(response.getEntity()));
+            }
+        } finally {
+            httpclient.close();
+        }
+    }
+
+    private void doTestGetCustomer(String contextUri, boolean withProcessor) throws Exception {
+        invokeGetCustomer("http://localhost:" + CXT + "/" + contextUri + "/customerservice/customers/126",
+                          "{\"Customer\":{\"id\":126,\"name\":\"CKC\"}}",
+                          withProcessor);
+        invokeGetCustomer("http://localhost:" + CXT + "/" + contextUri + "/customerservice/customers/123",
+                          "customer response back!",
+                          withProcessor);
+        invokeGetCustomer("http://localhost:" + CXT + "/" + contextUri + "/customerservice/customers/400",
+            "The remoteAddress is 127.0.0.1",
+            withProcessor);
+
+    }
+
+
+    public abstract static class AbstractTestProcessor implements Processor {
```

Comment:
I'd probably look into moving these processing code to a separate Camel-based test service (in a similar way to the one we have in the syslog tests, for example). That way, we can potentially reuse more test code when adding new scenarios to this.